### PR TITLE
Add Parser Strategy Tests

### PR DESCRIPTION
**Pull request by mrmrinal**
Add Tests for classes extending from ```ParserStrategy```

Also fixes the bug where an error showed when no output was shown when an Asset with no remarks was appended
#162

### CODE REVIEW

File: src/main/java/donnafin/logic/parser/AppendCommandParser.java [hunk 1]
```diff
@@ -91,7 +91,12 @@ private AppendCommand parseAsset(ArgumentMultimap argMultimap) throws ParseExcep
             throw new ParseException(Asset.MESSAGE_CONSTRAINTS);
         }
 
-        Asset newAsset = new Asset(name, type, value.toString(), remarks);
+        Asset newAsset = null;
+        try {
+            newAsset = new Asset(name, type, value.toString(), remarks);
+        } catch (IllegalArgumentException e) {
+            throw new ParseException(e.getMessage());
+        }
```
Comment (bharathcs) [inline src/main/java/donnafin/logic/parser/AppendCommandParser.java:99]: Not sure I understand your reasoning for this, would be helpful to add comments here to explain!

File: src/test/java/donnafin/logic/parser/AddressBookParserTest.java [hunk 2]
```diff
@@ -42,6 +43,11 @@ public void parseCommand_clear() throws Exception {
         assertTrue(parser.parseCommand(ClearCommand.COMMAND_WORD) instanceof ClearCommand);
     }
 
+    @Test
+    public void parseCommand_clearThrowsException_invalidInput() {
+        assertThrows(ParseException.class, () -> parser.parseCommand(ClearCommand.COMMAND_WORD, "25"));
```
Comment (bharathcs) [inline src/test/java/donnafin/logic/parser/AddressBookParserTest.java:48]: Could you also try other arguments like words and empty string:

e.g. `(ClearCommand.COMMAND_WORD, "")`, `(ClearCommand.COMMAND_WORD, "apple")`. These would be good equivalence partitions :)

File: src/test/java/donnafin/logic/parser/LiabilitiesTabParserTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                "10% annual interest");
+        assertTrue(model.getFilteredPersonList().get(0).getLiabilities().contains(addedLiability));
+    }
+
+    @Test
+    public void parserCommand_appendCommandInvalidInput_throwsException() {
+        assertThrows(ParseException.class, () -> parser.tabSpecificHandler("append",
+                "append ty/debt v/$100000 r/10% annual interest").execute(model));
+        assertThrows(ParseException.class, () -> parser.tabSpecificHandler("append",
+                "append n/Property debt with DBS v/$100000 r/10% annual interest").execute(model));
```
Comment (bharathcs) [inline src/test/java/donnafin/logic/parser/LiabilitiesTabParserTest.java:50]: Would you like to use AppendCommand.COMMAND_WORD here instead? Same with other instances of magic strings like "remove", "delete" in this file and other files.

**Review by bharathcs — COMMENTED** [review]
Nice work, Mrinal!! These are good test, but there are some points I raised about code quality, pls take a look.

File: src/main/java/donnafin/logic/parser/AppendCommandParser.java (hunk 1 above)
Comment (mrmrinal) [reply src/main/java/donnafin/logic/parser/AppendCommandParser.java:99]: This was a bug fix for the IllegalArgumentException caused and the output was not shown when a new Asset with no remarks was appended.

This edit ensures that a ParseException is thrown when a remark is not added to the appended asset

Here is the output displayed
image

Comment (bharathcs) [reply src/main/java/donnafin/logic/parser/AppendCommandParser.java:99]: Nice, thanks.

File: src/test/java/donnafin/logic/parser/AddressBookParserTest.java [hunk 4]
```diff
@@ -1,30 +1,21 @@
...
-import org.junit.jupiter.api.Test;
-
-import donnafin.logic.commands.AddCommand;
-import donnafin.logic.commands.ClearCommand;
-import donnafin.logic.commands.DeleteCommand;
-import donnafin.logic.commands.ExitCommand;
-import donnafin.logic.commands.FindCommand;
-import donnafin.logic.commands.HelpCommand;
-import donnafin.logic.commands.ListCommand;
+import donnafin.logic.commands.*;
```
Comment (bharathcs) [inline src/test/java/donnafin/logic/parser/AddressBookParserTest.java:3]: I'm afraid this wouldn't pass check style right? And there are other files where you use wildcard imports too.

File: src/test/java/donnafin/logic/parser/AssetsTabParserTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+import donnafin.model.ModelManager;
+import donnafin.model.UserPrefs;
+import donnafin.model.person.Asset;
+import donnafin.model.person.Person;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static donnafin.testutil.TypicalPersons.getTypicalAddressBook;
+import static donnafin.testutil.TypicalPersons.getTypicalPersons;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (bharathcs) [inline src/test/java/donnafin/logic/parser/AssetsTabParserTest.java:20]: Same here.

**Review by bharathcs — APPROVED** [review]
Nice work! But checkstyle doesn't pass... :(

**Review by bharathcs — APPROVED** [review]
LGTM!

File: src/test/java/donnafin/logic/parser/AddressBookParserTest.java (hunk 4 above)
Comment (mrmrinal) [reply src/test/java/donnafin/logic/parser/AddressBookParserTest.java]: My bad was working on my desktop , which did not allow me to change the import order statements. Have made a new commit that Appeases the checkstyle, and hopefully appeases you!

**Comment by bharathcs** [comment]
@mrmrinal you'd still need to merge master tho

**Pull request merged**